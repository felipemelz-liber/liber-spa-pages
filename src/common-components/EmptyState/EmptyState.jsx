import React from 'react';
import PropTypes from 'prop-types';
import Envelope from '../../../../assets/images/empty-states/envelope.png';
import Background from '../../../../assets/images/empty-states/background.png';
import {
  Content,
  LeftImage,
  TextContainer,
  Header,
  Paragraph,
  RightImage,
} from './EmptyState.styles';

const EmptyState = ({ header, text, imagePath }) => {
  return (
    <Content backgroundPath={Background}>
      <LeftImage src={imagePath} />
      <TextContainer>
        <Header>{header}</Header>
        <Paragraph>{text}</Paragraph>
      </TextContainer>
      <RightImage src={Envelope} />
    </Content>
  );
};

EmptyState.propTypes = {
  header: PropTypes.string,
  text: PropTypes.string,
  imagePath: PropTypes.string,
};

EmptyState.defaultProps = {
  header: '',
  text: '',
  imagePath: '',
};

export default EmptyState;
