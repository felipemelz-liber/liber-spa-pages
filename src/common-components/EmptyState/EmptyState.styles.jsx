import styled from 'styled-components';

export const Content = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  min-height: 220px;
  background-size: 100% 100%;
  background-image: url("${props => props.backgroundPath}");
  margin-top: 16px;
`;

export const LeftImage = styled.img`
  min-height: 225px;
  max-height: 225px;
`;

export const RightImage = styled.img`
  margin-bottom: -20px;
  min-height: 225px;
  max-height: 225px;
`;

export const TextContainer = styled.div`
  text-align: center;
  flex-grow: 1;
  margin: 0 60px 30px 60px;
`;

export const Header = styled.h1`
  font-family: Roboto;
  font-size: 22px;
  font-weight: 500;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.27;
  letter-spacing: 0.35px;
  text-align: center;
  color: #405f71;
`;

export const Paragraph = styled.div`
  font-family: Roboto;
  font-size: 16px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.38;
  letter-spacing: -0.38px;
  text-align: center;
  color: #798e9b;
`;
