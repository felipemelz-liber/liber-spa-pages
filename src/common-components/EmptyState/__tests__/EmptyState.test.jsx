import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import EmptyState from '../EmptyState';

describe('EmptyState tests', () => {
  configure({ adapter: new Adapter() });

  const props = {
    header: 'mock',
    text: 'mock',
    imagePath: 'mock',
  };

  it('should render correctly without props', () => {
    const wrapper = shallow(<EmptyState />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly with props', () => {
    const wrapper = shallow(<EmptyState {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
