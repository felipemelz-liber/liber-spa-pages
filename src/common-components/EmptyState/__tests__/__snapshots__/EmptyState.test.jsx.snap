// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState tests should render correctly with props 1`] = `
<styled.div
  backgroundPath="background.png"
>
  <styled.img
    src="mock"
  />
  <styled.div>
    <styled.h1>
      mock
    </styled.h1>
    <styled.div>
      mock
    </styled.div>
  </styled.div>
  <styled.img
    src="envelope.png"
  />
</styled.div>
`;

exports[`EmptyState tests should render correctly without props 1`] = `
<styled.div
  backgroundPath="background.png"
>
  <styled.img
    src=""
  />
  <styled.div>
    <styled.h1 />
    <styled.div />
  </styled.div>
  <styled.img
    src="envelope.png"
  />
</styled.div>
`;
